mender.go

```diff
@@ -164,6 +244,52 @@ func (m mender) GetDaemonConfig() daemonConfig {
 	}
 }
 
+func (m *mender) ForceBootstrap() {
+	m.forceBootstrap = true
+}
+
+func (m *mender) needsBootstrap() bool {
+	if m.forceBootstrap {
+		return true
+	}
+
+	if m.deviceKey.Private() == nil {
+		log.Debugf("needs keys")
+		return true
+	}
+
+	return false
+}
+
+func (m *mender) Bootstrap() error {
+	if !m.needsBootstrap() {
+		return nil
+	}
+
+	if m.deviceKey.Private() == nil {
+		log.Infof("device keys not present, generating")
+		if err := m.deviceKey.Generate(); err != nil {
+			return err
+		}
+
+		if err := m.deviceKey.Save(m.config.DeviceKey); err != nil {
+			log.Errorf("faiiled to save keys to %s: %s",
+				m.config.DeviceKey, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *mender) doBootstrap() error {
+
+	return nil
+}
```

Comment:
Where we are actually going to bootstrap, and what bootstrapping really means? Should we send some data to server and verify it accepts us (at least check that TLS connection was successful)?